src/components/Page.tsx

```diff
@@ -1,8 +1,37 @@
-import React, { Component } from "react";
+import PropTypes, { ValidationMap } from "prop-types";
+import React, { Attributes, ComponentType } from "react";
+
+export interface PageComponent<P = {}> {
+  id: string;
+  Component: string | ComponentType<Attributes | P>;
+  props?: P;
+}
+
+export interface PageProps<P = {}> {
+  components: PageComponent<P>[];
+}
+
+export class Page<P> extends React.Component<PageProps<P>> {
+  static propTypes: ValidationMap<PageProps> = {
+    components: PropTypes.arrayOf(
+      PropTypes.shape({
+        id: PropTypes.string.isRequired,
+        Component: PropTypes.elementType.isRequired,
+        props: PropTypes.object
+      }).isRequired
+    ).isRequired
+  };
 
-class Page extends Component {
   render(): JSX.Element {
-    return <div>Content goes here!</div>;
+    const { components } = this.props;
+
+    return (
+      <>
+        {components.map(({ id, Component, props }) => (
+          <Component key={id} {...props} />
```

Comment:
I didn't realise you could use a variable `Component` in this way to just load the right component, that's great.

Reply:
Yup. As long as your variable is PascalCase, you can do this. If it's camelCase, React looks for it as a plain HTML tag.

Note that `Component` can also equal `img` or `a` if you want to use basic HTML tags.